### PR TITLE
add http2-debug-state document

### PR DESCRIPTION
**Pull request by i110**
Added a document of http2-debug-state feature.

### CODE REVIEW

File: srcdoc/configure/http2_directives.mt [hunk 1]
```diff
@@ -273,4 +273,42 @@ Technically speaking, it does the following:
...
+</p>
+
+<p>
+The value of this directive specifies the property set contained in the response. Available values are <code>minimum</code> or <code>hpack</code>.
+If <code>hpack</code> is specified, the response will contain the internal hpack state of the same connection.
+If <code>minimum</code> is specified, the response doesn't contain the internal hpack state.
+</p>
+
+<p>
+In some circumstances, there may be a risk of information leekage on providing an internal hpack state. For example, the case that some proxies exist between the client and the server, and they share the connections among the clients.
```
Comment (syohex) [inline srcdoc/configure/http2_directives.mt]: `leekage` -> `leakage`

File: srcdoc/configure/http2_directives.mt [hunk 2]
```diff
@@ -273,4 +273,42 @@ Technically speaking, it does the following:
...
+    desc    => <<"EOT",
+A string enum value used to enable <a href=\"$spec_url\">HTTP/2 Implementation Debug State</a> feature and specify the property set of the response.
+EOT
+)->(sub {
+?>
+
+<p>
+This experimental feature serves a JSON document at the fixed path <code>/.well-known/h2/state</code>, which describes an internal HTTP/2 state of the H2O server.
+To know the details about the response fields, please see <a href=\"$spec_url\">the spec</a>.
+This feature is only for developping and debugging use, so it's highly recommended that you disable this setting in the production environment.
```
Comment (syohex) [inline srcdoc/configure/http2_directives.mt]: `developping` -> `developing`

File: srcdoc/configure/http2_directives.mt [hunk 3]
```diff
@@ -273,4 +273,42 @@ Technically speaking, it does the following:
...
+<?
+my $spec_url = "https://tools.ietf.org/html/draft-benfield-http2-debug-state-01";
+$ctx->{directive}->(
+    name    => "http2-debug-state",
+    levels  => [ qw(host) ],
+    see_also => render_mt(<<"EOT"),
+<a href=\"$spec_url\">HTTP/2 Implementation Debug State (draft-01)</a>
+EOT
+    desc    => <<"EOT",
+A string enum value used to enable <a href=\"$spec_url\">HTTP/2 Implementation Debug State</a> feature and specify the property set of the response.
```
Comment (kazuho) [inline srcdoc/configure/http2_directives.mt]: 細かな説明は後にあるのだから、desc には "A directive to turn on the HTTP/2 Implementation Debug State." とのみ書いたほうがよりシンプルになるかもしれません。

**Comment by kazuho** [comment]
Thank you for the PR. Please list the directives in ascending order, as well as taking a look at my in-line comment.

LGTM aside from the two.

**Comment by i110** [comment]
@kazuho @syohex Thank you for your comments. I fixed typos and other issues.

**Comment by kazuho** [comment]
Thank you for the docs!

**Comment by kazuho** [comment]
was part of #1019

**Pull request merged**